contents/architecture/RESTArchitecturalStyle.md

```diff
@@ -0,0 +1,226 @@
+- Internet Scale:
+  - Anarchic scalability: There is no central entity to control the whole system. Hence, firstly, the Web elements should be able to cope with unexpected load. As a result, it is not feasible for the clients to keep the knowledge about all the servers, and for the servers to keep track of the state of information at each of their clients. It is also costly to keep &quot;back-pointers&quot; inside the hypermedia data elements to refer back to the elements which reference them, as potentially there can be a huge number of references. Secondly, the Web elements should be able to ensure security against malformed or maliciously structured, which requires the Web to be able to &quot;communicate authentication data and authorization controls&quot;.
+  - Independent Deployment: to cater to the needs of different users, the Web has to allow the co-existence of both old and new implementations of the Web elements and undergo &quot;gradual and fragmented change&quot;. Hence the design of the Web should be able to ease the deployment of the Web elements &quot;in a partial, iterative fashion, since it is not possible to force deployment in an orderly manner&quot;.
+
+**REST architectural style**
+
+Fielding defined architectural style as &quot;a coordinated set of architectural constraints that restricts the roles/features of architectural elements and the allowed relationships among those elements within any architecture that conforms to that style&quot;.
+
+After surveying some common architectures for network-based applications on how the architectural constraints induce their corresponding architectural properties, Fielding came up with a new architectural style – Representational State Transfer, which satisfies the requirements of the Web development.
+
+The table below describes the REST architectural constraints and the architectural properties these constraints induce. Detailed information about the architectural properties can be found in **Appendix**.
+
+<table>
+	<tr>
+		<th>REST Constraint (Requirements satisfied)</th>
+		<th>Explanation</th>
+		<th>Properties induced/enhanced</th>
+	</tr>
+	<tr>
+		<td>
+			<p>Client-Server</p>
+			<p>(Extensibility)</p>
+		</td>
+		<td>
+			Separation of concerns: separate the user interface concern from the data storage concern  
+		</td>
+		<td>
+			<ul>
+				<li>
+					Portability of user interface across platforms which might differ in their data storage structures
+				</li>
+				<li>
+					Free the server from managing the user interface =&gt; simplified server component =&gt; enhanced scalability
```

Comment:
`=>` introduced here is a little confusing. It could mean an arrow, or it could mean greater than (in this context, more important than), and I think as computing students we are prone to think it is used as a greater than or equals to operator. :arrow_right: find this symbol [here](https://gist.github.com/rxaviers/7360908)